specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,293 @@
+
+**nimble-angular**  
+Each i18n provider will have its own Angular directive and module (prototype: [nimble-i18n-core.directive](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/ni/nimble-angular/src/directives/i18n/core/nimble-i18n-core.directive.ts) and [nimble-i18n-core.module](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/ni/nimble-angular/src/directives/i18n/core/nimble-i18n-core.module.ts) for `i18n-core`.)
+
+In order to make it easy/automatic for clients to pick up new localized strings/labels when they uptake new nimble-angular versions, each i18n provider has an additional directive that will set all of the Nimble-defined labels/strings, using Angular's `$localize` function on the English strings.  
+Prototype: [nimble-i18n-core-with-defaults.directive](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/ni/nimble-angular/src/directives/i18n/core/nimble-i18n-core-with-defaults.directive.ts)
+
+For each i18n provider that an Angular app will use:
+- The app imports that specific i18n module (prototype: [in example app module](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/example-client-app/src/app/app.module.ts#L73)). This will result in the app pulling in the Nimble-provided labels/strings for localization when they run `ng extract-i18n`, due to the `with-defaults.directive` using `$localize`. When they pull in new nimble-angular versions in the future, new strings in the same i18n provider will automatically get pulled in too. (Prototype: [messages.xlf](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/example-client-app/src/locales/messages.xlf), output of `ng extract-i18n` after importing Nimble i18n core module)
+- The app adds that i18n element as a child to their theme provider ([prototype](https://github.com/ni/nimble/blob/d51ee14dc49db7070e5cab726c225f69635de17b/angular-workspace/projects/example-client-app/src/app/app.component.html#L2)):
+```html
+<nimble-theme-provider>
+  <nimble-i18n-core withDefaults></nimble-i18n-core>
+</nimble-theme-provider>
+```
+- If the app needs to customize any of the labels, they can do so via the i18n directive API. Generally the root i18n provider would use `withDefaults` to set all the labels to their localized values, and any nested ones would not.
+
+Codegen: We should be able to codegen the Angular directives. The prototype does not, for simplicity.
```

Comment:
Looking at the prototype, I'm siding with Milan's opinion that we should start by not code generating. The number of strings is small enough that it doesn't seem worth it.

Reply:
I guess the main concern there would be that the English strings (and the descriptions if we have them) have to be copy/pasted into the Angular directives too, so it's possible they could end up out-of-sync with what's in nimble-components.

Since `$localize` expects a templated string as input, I didn't see a way to have nimble-components export the English string values such that they could be used with `$localize` without codegen.

Reply:
Ah, great point. That seems compelling to me; let's mention it in the spec as part of the rationale for code gen.

Reply:
I'm skeptical that is a major concern for the same reasons. I don't think those localized strings are changing frequently and I don't think there are so many that it is a significant challenge to keep in sync. Also they are placeholders / defaults. The whole point is that localization is going to replace them all anyway.

If the consensus is still that generation is useful then I still want to discuss how we minimize the amount needed.

Reply:
Updated the wording about codegen to include the pro (not duplicating the English string) and con (obfuscating the code a bit). Resolving this thread as I don't think we need to definitively commit to one or the other in this HLD.

Reply:
I'm ok with deferring this decision until implementation. But in the spirit of the "large long-lasting PRs" conversation we had today, it'd be good to try to resolve it early with a prototype PR or something rather than completing the entire implementation with one approach only to have us decide to go with the other.